src/binaryutils.cpp

```diff
@@ -299,7 +299,8 @@ InferenceEngine::Blob::Ptr createBlobFromMats(const std::vector<cv::Mat>& images
 }
 
 ov::runtime::Tensor createBlobFromMats_2(const std::vector<cv::Mat>& images, const std::shared_ptr<TensorInfo>& tensorInfo, bool isPipeline) {
-    ov::Shape shape = tensorInfo->getShape();
+    auto dims = !isPipeline ? tensorInfo->getShape() : getShapeFromImages(images, tensorInfo);
```

Comment:
Why do we need tensorInfo to get shape from images? If this is from images, we should be possible to get it without additional info.

Reply:
For security reasons we should stick to the tensorInfo if possible. With binary inputs it is sometimes not possible to fully validate input at this stage since we had dynamic shapes in DAGs.